ViewEnvironmentUI/Tests/ViewEnvironmentObservingTests.swift

```diff
@@ -324,6 +324,70 @@ final class ViewEnvironmentObservingTests: XCTestCase {
         container.setNeedsEnvironmentUpdate()
         XCTAssertEqual(observedEnvironments.count, 2)
     }
+
+    // MARK: - Customizations
+
+    func test_customization() throws {
+        let viewController = TestViewEnvironmentObservingViewController()
+
+        // Customizations should be respected.
+        do {
+            var customizationLifetime: ViewEnvironmentCustomizationLifetime? = viewController
+                .addEnvironmentCustomization {
+                    $0.testContext.number = 200
+                }
+
+            XCTAssertEqual(viewController.environment.testContext.number, 200)
+
+            _ = customizationLifetime // Suppress warning about variable never being read
```

Comment:
Wrapping up the lifetime's scope in `withExtendedLifetime` is another way.

Reply:
Agree and I do prefer that, but this is the pattern used in the other tests in this module so I did the same here. I can migrate them all in this PR if you think it's worth the slightly increased scope—let me know!

Reply:
Nevermind, I'll just update these now, there's only 1!

Reply:
a693580f